lib/db/mem.js

```diff
@@ -251,14 +294,28 @@ module.exports = function (log, error) {
 
   Memory.prototype.accountDevices = function (uid) {
     return getAccountByUid(uid)
-      .then(function(account) {
-        var devices = Object.keys(account.devices).map(
-          function (id) {
-            return account.devices[id]
-          }
-        )
-        return P.resolve(devices)
-      })
+      .then(
+        function(account) {
+          return Object.keys(account.devices).map(
+            function (id) {
+              var device = devices[internalDeviceId(account, { id: id })]
+              var session = sessionTokens[device.sessionTokenId]
+              if (session) {
+                device.uaBrowser = session.uaBrowser
+                device.uaBrowserVersion = session.uaBrowserVersion
+                device.uaOS = session.uaOS
+                device.uaOSVersion = session.uaOSVersion
+                device.uaDeviceType = session.uaDeviceType
+                device.lastAccessTime = session.lastAccessTime
+              }
+              return device
+            }
+          )
+        },
+        function (err) {
+          return []
+        }
+      )
```

Comment:
listing sessions is already another thing https://github.com/mozilla/fxa-auth-db-mysql/blob/master/lib/db/mysql.js#L317-L321